File: factory/src/main/java/my/sample/factory/impl/InternalBean.java
package my.sample.factory.impl;

import java.util.Map;

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class InternalBean {
    private String foo;
    private String bar;
    private String baz;

    public InternalBean() {
        System.out.println("new InternalBean: " + this);
    }

    public String echo(String message) {
        return "Echo processed: " + message;
    }

    public void update(Map<String, ?> config) {
        System.out.println("updating....: " + this);
    }

    public void setFoo(String foo) {
        System.out.println("setting new foo:" + foo);
        this.foo = foo;
    }

    public void setBar(String bar) {
        System.out.println("setting new bar:" + bar);
        this.bar = bar;
    }

    public void setBaz(String baz) {
        System.out.println("setting new baz:" + baz);
        this.baz = baz;
    }

    @Override
    public String toString() {
        return " InternalBean {" +
                " foo = '" + foo + '\'' +
                ", bar = '" + bar + '\'' +
                ", baz = '" + baz + '\'' +
                " }";
    }
}

File: factory/src/main/kotlin/my/sample/factory/impl/InternalBean.kt
package my.sample.factory.impl

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class InternalBean {
    private var foo: String? = null
    private var bar: String? = null
    private var baz: String? = null
    {
        println("new InternalBean: $this")
    }

    public fun echo(message: String): String = "Echo processed: $message"

    public fun update(config: Map<String, *> ) {
        println("updating....: $this")
    }

    public fun setFoo(foo: String) {
        println("setting new foo: $foo")
        this.foo = foo
    }

    public fun setBar(bar: String) {
        println("setting new bar: $bar")
        this.bar = bar
    }

    public fun setBaz(baz: String) {
        println("setting new baz: $baz")
        this.baz = baz
    }

    override public fun toString() = " InternalBean { foo = '$foo' , bar = '$bar', baz = '$baz' }"
}
